utilities/src/main/java/io/onetable/utilities/RunSync.java

```diff
@@ -148,24 +150,44 @@ public static void main(String[] args) throws IOException {
           "Running sync for basePath {} for following table formats {}",
           table.getTableBasePath(),
           tableFormatList);
-      PerTableConfig config =
-          PerTableConfigImpl.builder()
-              .tableBasePath(table.getTableBasePath())
-              .tableName(table.getTableName())
+      SourceTable sourceTable =
+          SourceTable.builder()
+              .name(table.getTableName())
+              .basePath(table.getTableBasePath())
               .namespace(table.getNamespace() == null ? null : table.getNamespace().split("\\."))
-              .tableDataPath(table.getTableDataPath())
-              .icebergCatalogConfig(icebergCatalogConfig)
-              .hudiSourceConfig(
-                  HudiSourceConfigImpl.builder()
-                      .partitionSpecExtractorClass(
-                          ConfigurationBasedPartitionSpecExtractor.class.getName())
-                      .partitionFieldSpecConfig(table.getPartitionSpec())
-                      .build())
-              .targetTableFormats(tableFormatList)
+              .dataPath(table.getTableDataPath())
+              .catalogConfig(icebergCatalogConfig)
+              .formatName(sourceFormat)
+              .build();
+
+      List<TargetTable> targetTables =
```

Comment:
I'm wondering if anyone has ideas on how to make the setup less verbose. The construction of the source and the targets share many of the same properties so it seems like there should be a simpler way of doing this.

Reply:
Could the parent class, `ExternalTable`, be used for this along with a copy constructor. i.e. SourceTable.from(ExternalTable table)?